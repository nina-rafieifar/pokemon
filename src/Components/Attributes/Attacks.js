import { html, LitElement, css } from "lit-element";
class PokemonAttacks extends LitElement {
  static get properties() {
    return {
      name: { attribute: true },
      damage: { attribute: true }
    };
  }
  static get styles() {
    return css`
      .chip {
        display: inline-block;
        background: #14855a;
        padding: 0 12px;
        border-radius: 5px;
        font-size: 13px;
        font-family: Space Mono;
        display: inline-flex;
        justify-content: space-around;
        min-width: 6rem;
        color: #ffffff;
        width: max-content;
      }
      .pokemon__details_item {
        padding: 0.15rem;
        height: 1.4rem;
      }
    `;
  }
  render() {
    return html`<div class="pokemon__details_item">
      <div class="chip">
        <span>${this.name}</span>
        |
        <span>${this.damage}</span>
      </div>
    </div>`;
  }
}

customElements.define("pokemon-attacks", PokemonAttacks);
