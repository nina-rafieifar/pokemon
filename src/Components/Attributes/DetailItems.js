import { html } from "lit-element";

export const PokemonWeightAndWeight = (text, Max, Min) => html`<div class="pokemon__details_items">
  <span class="size__text">${text}</span>
  <div class="size">
    <span>${Min} - ${Max}</span>
  </div>
</div>`;

export const PokemonResistantAndWeaknesses = (text, ability) => html`<div class="pokemon__details_items">
  <span class="chip__text">${text}</span>
  <div class="chips">
    ${ability.split(",").map(
      (i) => html`<div class="chip chip_${i.toLowerCase()}">
        <span>${i}</span>
      </div>`
    )}
  </div>
</div>`;
