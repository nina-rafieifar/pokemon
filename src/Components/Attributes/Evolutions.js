import { html, LitElement, css } from "lit-element";
class PokemonEvolution extends LitElement {
  static get properties() {
    return {
      name: { attribute: true },
      number: { attribute: true },
      image: { attribute: true }
    };
  }
  static get styles() {
    return css`
      .pokemon__evolution_general-attributes {
        box-shadow: 0 8px 16px 0 rgba(0, 0, 0, 0.2);
        transition: 0.3s;
        border-radius: 50%;
        background-color: #ffffff;
        border: 1px solid #3b3939;
        height: 25vh;
        width: 25vh;
        color: #3b3939;
      }
      .pokemon__evolution_name {
        margin: 0;
        margin: -25px;
        font-size: small;
      }
      .pokemon__evolution_number {
        padding: 0.4rem 0px;
        font-family: Tigerious;
        margin: 0.1rem;
      }
      .pokemon__evolution_image {
        height: 12vh;
        pointer-events: none;
        text-align: left;
      }
    `;
  }
  render() {
    return html`<div class="pokemon__evolution_general-attributes">
      <p class="pokemon__evolution_number">${this.number}</p>
      <img alt=${this.name} src=${this.image} class="pokemon__evolution_image" />
      <h4 class="pokemon__evolution_name">${this.name}</h4>
    </div>`;
  }
}

customElements.define("pokemon-evolution", PokemonEvolution);
