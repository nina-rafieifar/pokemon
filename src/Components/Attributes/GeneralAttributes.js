import { html, LitElement, css } from "lit-element";

class PokemonAttr extends LitElement {
  static get properties() {
    return {
      name: { attribute: true },
      number: { attribute: true },
      image: { attribute: true }
    };
  }
  static get styles() {
    return css`
      .pokemon__general-attributes {
        box-shadow: 0 16px 32px 0 rgba(0, 0, 0, 0.2);
        transition: 0.3s;
        border-radius: 22px;
        background-color: #ffffff;
        max-height: 87vh;
      }
      .pokemon__general-attributes:hover {
        box-shadow: 0 35px 45px 0 rgba(0, 0, 0, 0.2);
      }
      .pokemon__name {
        margin: 0;
      }
      .pokemon__number {
        padding: 1rem 0;
        font-family: Tigerious;
      }
      .pokemon__image {
        height: 50vh;
        pointer-events: none;
        text-align: left;
      }
    `;
  }
  render() {
    return html`<div class="pokemon__general-attributes">
      <h2 class="pokemon__number">${this.number}</h2>
      <img alt=${this.name} src=${this.image} class="pokemon__image" />
      <h1 class="pokemon__name">${this.name}</h1>
    </div>`;
  }
}

customElements.define("pokemon-general-attr", PokemonAttr);
