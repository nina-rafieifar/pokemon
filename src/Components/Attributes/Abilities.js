import { html, LitElement, css } from "lit-element";
import { PokemonResistantAndWeaknesses } from "./DetailItems";
class PokemonAbilities extends LitElement {
  static get properties() {
    return {
      resistant: { attribute: true },
      weaknesses: { attribute: true }
    };
  }
  static get styles() {
    return css`
      .chips {
        display: grid;
      }
      .chip {
        display: inline-block;
        background: #919191;
        padding: 0 12px;
        border-radius: 5px;
        font-size: 13px;
        font-family: Space Mono;
        margin: 4%;
        color: #ffffff;
      }
      .chip_bug {
        background: #27a852;
      }
      .chip_dark {
        background: #30303b;
      }
      .chip_grass {
        background: #88cc00;
      }
      .chip_ghost {
        color: #000000;
        background: #e8e8ff;
      }
      .chip_ground {
        background: #919191;
        background: linear-gradient(to top, #875850 50%, #919191 50%);
      }
      .chip_fairy {
        color: #000000;
        background: #ffd1e8;
        background: linear-gradient(to top, #8373ff 50%, #ffd1e8 50%);
      }
      .chip_fire {
        background: #ff0000bb;
      }
      .chip_flying {
        background: #56b8b0;
      }
      .chip_ice {
        color: #000000;
        background: #e6e6e6;
      }
      .chip_water {
        background: #3e86fa;
      }
      .chip__text {
        font-family: Tigerious;
      }
      .pokemon__details_resistance_weakness {
        display: flex;
        justify-content: space-around;
      }
    `;
  }
  render() {
    return html` <div class="pokemon__details_resistance_weakness">
      ${PokemonResistantAndWeaknesses("Resistances", this.resistant)}${PokemonResistantAndWeaknesses(
        "Weaknesses",
        this.weaknesses
      )}
    </div>`;
  }
}

customElements.define("pokemon-abilities", PokemonAbilities);
