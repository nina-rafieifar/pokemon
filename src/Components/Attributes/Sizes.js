import { html, LitElement, css } from "lit-element";
import { PokemonWeightAndWeight } from "./DetailItems";
class PokemonSizes extends LitElement {
  static get properties() {
    return {
      weightMax: { attribute: true },
      weightMin: { attribute: true },
      heightMax: { attribute: true },
      heightMin: { attribute: true }
    };
  }
  static get styles() {
    return css`
      .pokemon__details_sizes {
        display: flex;
        font-family: Tigerious;
      }
      .pokemon__details_items {
      }
      .size__text {
      }
      .size {
        background: #3524b5;
        padding: 0px 12px;
        border-radius: 5px;
        font-size: 13px;
        font-family: "Space Mono";
        display: inline-flex;
        min-width: 70%;
        justify-content: space-around;
        margin: 0.3rem 0;
        color: #ffffff;
      }
    `;
  }
  render() {
    return html`<div class="pokemon__details_sizes">
      ${PokemonWeightAndWeight("Weight", this.weightMax, this.weightMin)}
      ${PokemonWeightAndWeight("Height", this.heightMax, this.heightMax)}
    </div>`;
  }
}

customElements.define("pokemon-sizes", PokemonSizes);
