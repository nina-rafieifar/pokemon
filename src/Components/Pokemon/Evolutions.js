import React from "react";
import KeyboardArrowRightIcon from "@material-ui/icons/KeyboardArrowRight";
import "../Attributes/Evolutions";
import "../Attributes/Sizes";
import "./style.scss";

export function Evolutions(prop) {
  return (
    <table className="pokemon__evolutions_table">
      <tbody>
        <tr>
          <td>
            <div className="pokemon__evolutions_container">
              <p>Evolution</p>
              {prop.pokemon.evolutions ? (
                <div className="pokemon__evolutions_items">
                  <pokemon-evolution
                    name={prop.pokemon.name}
                    number={prop.pokemon.number}
                    image={prop.pokemon.image}
                  ></pokemon-evolution>
                  <KeyboardArrowRightIcon />
                  <pokemon-evolution
                    name={prop.pokemon.evolutions[0].name}
                    number={prop.pokemon.evolutions[0].number}
                    image={prop.pokemon.evolutions[0].image}
                  ></pokemon-evolution>
                </div>
              ) : (
                <div className="pokemon__evolutions_items pokemon__evolutions_items_no-evolve">
                  <p>This Pok√©mon does not evolve.</p>
                  <div>
                    <pokemon-evolution
                      name={prop.pokemon.name}
                      number={prop.pokemon.number}
                      image={prop.pokemon.image}
                    ></pokemon-evolution>
                  </div>
                </div>
              )}
            </div>
          </td>
        </tr>
        <tr>
          <td className="pokemon__details_items">
            <pokemon-sizes
              weightMax={prop.pokemon.weight.maximum}
              weightMin={prop.pokemon.weight.minimum}
              heightMax={prop.pokemon.height.maximum}
              heightMin={prop.pokemon.height.minimum}
            ></pokemon-sizes>
          </td>
        </tr>
      </tbody>
    </table>
  );
}
