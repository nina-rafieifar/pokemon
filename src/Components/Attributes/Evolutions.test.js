import "../Attributes/Evolutions";
import { fixture } from "@open-wc/testing";
import { html, render } from "lit-html";

test("pokemon-evolution renders the correct elements", () => {
  render(html`<pokemon-evolution name="Ivysaur" number="002" image="some url"></pokemon-evolution>`, document.body);
  expect(document.body.getElementsByTagName("pokemon-evolution")).toBeDefined();
});

test("pokemon-evolution renders as it should", async () => {
  const element = await fixture(
    html`<pokemon-evolution name="Ivysaur" number="002" image="some url"></pokemon-evolution>`
  );
  expect(element).toMatchSnapshot();
});
