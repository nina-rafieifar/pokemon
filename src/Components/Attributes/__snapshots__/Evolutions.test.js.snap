// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pokemon-evolution renders as it should 1`] = `
<pokemon-evolution
  image="some url"
  name="Ivysaur"
  number="002"
/>
`;
